### PR TITLE
AIP-0231 Batch methods: Get

### PR DESCRIPTION
**Pull request by michaelwq**
1) implement AIP-0231 Batch methods: Get
2) import "github.com/kjk/inflect" to check the plural or singular form of a word

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/golang/protobuf v1.3.2
 	github.com/google/go-cmp v0.3.1
 	github.com/jhump/protoreflect v1.5.1-0.20191024142022-1d0024204e58
+	github.com/kjk/inflect v0.0.0-20190213232030-bfc9dcd6393b
```
Comment (mingzhi) [inline go.mod]: Can you use `github.com/gertd/go-pluralize` instead?

File: rules/aip0231/message_rules.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+		// Rule check: Ensure that the standard get request message field is repeated.
+		if getReqMsg != nil && !getReqMsg.IsRepeated() {
+			problems = append(problems, lint.Problem{
+				Message:    `The "requests" field should be repeated`,
+				Descriptor: getReqMsg,
+			})
+		}
+		// Rule check: Ensure that the standard get request message field is the correct type.
+		rightTypeName := fmt.Sprintf("Get%sRequest", inflect.ToSingular(m.GetName()[8:len(m.GetName())-7]))
```
Comment (mingzhi) [inline rules/aip0231/message_rules.go]: Consider to add a comment to explain `m.GetName()[8:len(m.GetName())-7])`? It took me some time to figure it out: `"BatchGetBooksRequest"` -> `"Books"`.

File: rules/aip0231/message_rules.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+// The Batch Get response message should have resource field.
+var resourceField = &lint.MessageRule{
+	Name:   lint.NewRuleName("core", "0231", "response-message", "resource-field"),
+	URI:    "https://aip.dev/231#response-message",
+	OnlyIf: isBatchGetResponseMessage,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// the singular form the resource name, the first letter is Capitalized
+		resourceMsgName := inflect.ToSingular(m.GetName()[8 : len(m.GetName())-9])
```
Comment (mingzhi) [inline rules/aip0231/message_rules.go]: Consider to add a comment to explain `m.GetName()[8 : len(m.GetName())-9]`.

File: rules/aip0231/message_rules.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+		for _, fieldDesc := range m.GetFields() {
+			if msgDesc := fieldDesc.GetMessageType(); msgDesc != nil && msgDesc.GetName() == resourceMsgName {
+				if !fieldDesc.IsRepeated() {
+					return []lint.Problem{{
+						Message:    fmt.Sprintf("The %q type field on Batch Get Response message should be repeated", msgDesc.GetName()),
+						Descriptor: fieldDesc,
+					}}
+				} else {
```
Comment (mingzhi) [inline rules/aip0231/message_rules.go]: Consider to remove `else`.

File: rules/aip0231/method_rules.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+// Batch Get method should have a properly named Request message.
+var inputName = &lint.MethodRule{
+	Name:   lint.NewRuleName("core", "0231", "input-message", "name"),
+	URI:    "https://aip.dev/231#guidance",
+	OnlyIf: isBatchGetMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Rule check: Establish that for methods such as `BatchGetFoos`, the request
+		// message is named `BatchGetFoosRequest`.\
```
Comment (mingzhi) [inline rules/aip0231/method_rules.go]: Can you remove the duplicated comments? Thanks!

File: rules/aip0231/method_rules.go [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+	},
+}
+
+// Batch Get method should have a properly named Response message.
+var outputName = &lint.MethodRule{
+	Name:   lint.NewRuleName("core", "0231", "output-message", "name"),
+	URI:    "https://aip.dev/231#guidance",
+	OnlyIf: isBatchGetMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Rule check: Establish that for methods such as `BatchGetFoos`, the response
```
Comment (mingzhi) [inline rules/aip0231/method_rules.go]: Can you remove the duplicated comments? Thanks!

File: go.mod (hunk 1 above)
Comment (lukesneeringer) [reply go.mod]: +1, please use the dependency we already have.

Comment (michaelwq) [reply go.mod]: Done.

File: rules/aip0231/message_rules.go (hunk 3 above)
Comment (michaelwq) [reply rules/aip0231/message_rules.go]: Done.

File: rules/aip0231/message_rules.go (hunk 4 above)
Comment (michaelwq) [reply rules/aip0231/message_rules.go]: Done.

File: rules/aip0231/method_rules.go (hunk 5 above)
Comment (michaelwq) [reply rules/aip0231/method_rules.go]: Done.

File: rules/aip0231/method_rules.go (hunk 6 above)
Comment (michaelwq) [reply rules/aip0231/method_rules.go]: Done.

File: rules/aip0231/aip0231.go [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+func isBatchGetRequestMessage(m *desc.MessageDescriptor) bool {
+	return batchGetReqMessageRegexp.MatchString(m.GetName())
+}
+
+// Returns true if this is an AIP-231 Get request message, false otherwise.
+func isBatchGetResponseMessage(m *desc.MessageDescriptor) bool {
+	return batchGetResMessageRegexp.MatchString(m.GetName())
+}
+
+//// get resource message type name from method
```
[the remark this answers is not in the document]
Comment (michaelwq) [reply rules/aip0231/aip0231.go]: Done.

**Pull request merged**